verilog/analysis/checkers/constraint_name_style_rule.cc

```diff
@@ -44,50 +42,64 @@ using verible::matcher::Matcher;
 // Register ConstraintNameStyleRule.
 VERILOG_REGISTER_LINT_RULE(ConstraintNameStyleRule);
 
-static constexpr absl::string_view kMessage =
-    "Constraint names must by styled with lower_snake_case and end with _c.";
-
-const LintRuleDescriptor& ConstraintNameStyleRule::GetDescriptor() {
+const LintRuleDescriptor &ConstraintNameStyleRule::GetDescriptor() {
   static const LintRuleDescriptor d{
       .name = "constraint-name-style",
       .topic = "constraints",
       .desc =
-          "Check that constraint names follow the lower_snake_case "
-          "convention and end with _c.",
+          "Check that constraint names follow the required name style "
+          "specified by a regular expression.",
+      .param = {{"pattern", kSuffix}},
   };
   return d;
 }
 
-static const Matcher& ConstraintMatcher() {
+absl::Status ConstraintNameStyleRule::Configure(
+    absl::string_view configuration) {
+  std::string pattern = kSuffix;
+  absl::Status status = verible::ParseNameValues(
+      configuration, {{"pattern", verible::config::SetString(&pattern)}});
+
+  regex = std::make_unique<re2::RE2>(pattern);
+
+  return status;
+}
+
+static const Matcher &ConstraintMatcher() {
   static const Matcher matcher(NodekConstraintDeclaration());
   return matcher;
 }
 
-void ConstraintNameStyleRule::HandleSymbol(const verible::Symbol& symbol,
-                                           const SyntaxTreeContext& context) {
+void ConstraintNameStyleRule::HandleSymbol(const verible::Symbol &symbol,
+                                           const SyntaxTreeContext &context) {
   verible::matcher::BoundSymbolManager manager;
   if (ConstraintMatcher().Matches(symbol, &manager)) {
     // Since an out-of-line definition is always followed by a forward
     // declaration somewhere else (in this case inside a class), we can just
-    // ignore all out-of-line definitions to  avoid duplicate lint errors on
+    // ignore all out-of-line definitions to avoid duplicate lint errors on
     // the same name.
     if (IsOutOfLineConstraintDefinition(symbol)) {
       return;
     }
 
-    const auto* identifier_token =
+    const auto *identifier_token =
         GetSymbolIdentifierFromConstraintDeclaration(symbol);
     if (!identifier_token) return;
 
     const absl::string_view constraint_name = identifier_token->text();
 
-    if (!verible::IsLowerSnakeCaseWithDigits(constraint_name) ||
-        !absl::EndsWith(constraint_name, "_c")) {
-      violations_.insert(LintViolation(*identifier_token, kMessage, context));
+    if (!RE2::FullMatch(constraint_name, *regex)) {
+      violations_.insert(
+          LintViolation(*identifier_token, FormatReason(), context));
```

Comment:
Didn't find many resources on RE2. Can it somehow help us provide a better linting message than just "follow the regex"? Perhaps provide an autofix?